### DISCUSSION TITLE
Pyvips - Gaussian Blur has black outline around image

### DISCUSSION THREAD

**Discussion by sairam-haribabu** [libvips/libvips#3851] [Q&A]
When performing gaussian blur with the setting below, the output image has a weird black outline around it. 

`image = image.gaussblur(sigma, min_ampl=0.1, precision="float")
`
Source image:
![cathub-resized](https://github.com/libvips/libvips/assets/89160461/160229ec-cbfb-4b08-a715-fd3375342c08)

Output (sigma = 10)
![cathub-blur-r-30](https://github.com/libvips/libvips/assets/89160461/8c71e8fb-140b-41bc-a8fc-211492f7197d)

I'm trying to find a fix for the black outline around the image. Hoping to see some output like this:

![9MB-Cathub-magick-gaussian-blur-1](https://github.com/libvips/libvips/assets/89160461/9f87c030-60b6-4091-8d22-a4134c0bf0f2)

**Environment**
(please complete the following information)
 - OS: MacOS 13.6
 - Vips: 8.15

PS: Issue observed best using the browser's light mode

TIA!

**Comment by jcupitt** [comment]
Hello @sairam-haribabu,

That's the edge of the image being blended into your PNG background, which is black. Try dropping the alpha:

```
vips extract_band whipped.png x.png 0 --n 3
```

ie. starting from band 0, extract three bands (bands are always RGBA) to make:

![x](https://github.com/libvips/libvips/assets/580843/13bdbed4-5555-4774-bf34-595a7f661c2c)

You can see your image is really on a black background, it's only the transparent alpha which makes it seem white. When you blur it, you are seeing these black pixels.

To fix this, flatten out the alpha against white to set the background pixels, reattach the alpha, then blur.

```python
fixed_rgba = image.flatten(background=255).bandjoin(image[3])
image = fixed_rgba.gaussblur(5)
```

To make:

![x](https://github.com/libvips/libvips/assets/580843/b8d29139-e1e7-4a40-8c38-26a5079f19f0)

You'll find it looks correct on a white and on a black background.
